### PR TITLE
Remove <p></p> around the IDs of preserved chunks

### PR DESCRIPTION
**Pull request by yihui**
e.g. see 

- https://github.com/rstudio/DT/blob/053a506d0/001-fixedheader.html#L160-L196
- https://github.com/rstudio/leaflet/blob/c92d6f1d0/basemaps.html#L141-L181

### CODE REVIEW

**Comment by jjallaire** [comment]
@jmcphers Could you review this?

**Comment by jmcphers** [comment]
Looks good.

File: R/html_document_base.R [hunk 1]
```diff
@@ -135,8 +135,15 @@ html_document_base <- function(smart = TRUE,
     output_str <- readLines(output_file, warn = FALSE, encoding = "UTF-8")
 
     # if we preserved chunks, restore them
-    if (length(preserved_chunks) > 0)
+    if (length(preserved_chunks) > 0) {
+      # Pandoc adds an empty <p></p> around the IDs of preserved chunks, and we
+      # need to remove these empty tags, otherwise we may have invalid HTML like
+      # <p><div>...</div></p>
+      for (i in names(preserved_chunks)) {
+        output_str <- gsub(paste0("<p>", i, "</p>"), i, output_str, fixed = TRUE)
```
Comment (jjallaire) [inline R/html_document_base.R]: I noticed that the call to gsub in restorePreservedChunks includes `useBytes = TRUE`. Do we need to do that as well here?

Comment (yihui) [inline R/html_document_base.R]: I see you have done it. Usually this is only relevant when the system locale is C and there are multibyte characters to be substituted. It is not harmful to use `useBytes = TRUE` anyway.

**Pull request merged**